### PR TITLE
docs: remove dead link to chromiumdev slack

### LINKED ISSUE #17154: Link chromiumdev-slack refers to some king of strange site
Did you see that link to chromiumdev-slack on https://electronjs.org/docs/development/chromium-development page refers to this site https://chromiumdev-slack.herokuapp.com/index.html which is not the slack related page?

### PR DESCRIPTION
**Pull request by nornagon**
The link seems to be showing something unrelated to chromium-dev slack.

Closes #17154 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: no-notes

### CODE REVIEW

File: docs/development/chromium-development.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 > A collection of resources for learning about Chromium and tracking its development
 
-- [chromiumdev](https://chromiumdev-slack.herokuapp.com) on Slack
+- [#chromium](irc://irc.freenode.net/chromium) on Freenode IRC
```
Comment (jkleinsc) [inline docs/development/chromium-development.md]: Our linter doesn't like the irc:// protocol.  Can we use a different link or remove the link?

**Comment by nornagon** [comment]
is lint borked?

**Comment by MarshallOfSound** [comment]
@nornagon All of CircleCI is borked, we should suspend merging

**Comment by codebytere** [comment]
@nornagon maybe just rebase and then we can merge?

**Pull request merged**